src/app/settings/action-triggers/components/action-triggers/action-triggers.component.ts

```diff
@@ -0,0 +1,100 @@
+import { Logger } from 'src/app/core/abstractions/logger.service'
+import { Component, OnDestroy, OnInit } from '@angular/core'
+import { ActionTrigger, ActionTriggerWithActionInfo } from 'src/app/shared/models/action-trigger'
+import { ActionTriggerStateService } from 'src/app/core/services/action-trigger-state.service'
+import { Subject, takeUntil } from 'rxjs'
+import { Tv2PartAction } from 'src/app/shared/models/tv2-action'
+import { ActionStateService } from 'src/app/shared/services/action-state.service'
+import { HttpFileDownloadService } from 'src/app/core/services/http/http-file-download.service'
+import { KeyboardTriggerData } from 'src/app/shared/models/keyboard-trigger'
+
+@Component({
+  selector: 'sofie-action-triggers',
+  templateUrl: './action-triggers.component.html',
+  styleUrls: ['./action-triggers.component.scss'],
+})
+export class ActionTriggersComponent implements OnInit, OnDestroy {
+  public selectedAction: ActionTriggerWithActionInfo<KeyboardTriggerData> | undefined
+  public createAction: boolean
+  public loading: boolean = true
+  public actions: Tv2PartAction[]
+  public actionTriggers: ActionTriggerWithActionInfo<KeyboardTriggerData>[]
+  private readonly unsubscribe$: Subject<null> = new Subject<null>()
+
+  constructor(
+    private readonly actionTriggerStateService: ActionTriggerStateService,
+    private readonly fileDownloadService: HttpFileDownloadService,
+    private readonly logger: Logger,
+    private readonly actionStateService: ActionStateService
+  ) {}
+
+  public ngOnInit(): void {
+    this.actionStateService
+      .subscribeToRundownActions()
+      .then(observable => observable.subscribe(actions => this.onActionsChanged(actions as Tv2PartAction[])))
+      .then(() => this.subscribeForActionTriggerObservable())
+      .catch(error => this.logger.data(error).error('Error while listening to Action events'))
+  }
+
+  private subscribeForActionTriggerObservable(): void {
+    this.actionTriggerStateService
+      .getActionTriggerObservable()
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe(triggers => {
+        this.actionTriggers = JSON.parse(
+          JSON.stringify(
+            (triggers as ActionTrigger<KeyboardTriggerData>[]).map(trigger => {
+              return { ...trigger, actionInfo: this.actions.find(action => action.id === trigger.actionId) }
+            })
+          )
+        )
+        this.loading = false
+      })
+  }
+
+  private onActionsChanged(loadedActions: Tv2PartAction[]): void {
+    this.actions = JSON.parse(JSON.stringify(loadedActions))
+    this.loading = false
+  }
+
+  public ngOnDestroy(): void {
+    this.unsubscribe$.next(null)
+    this.unsubscribe$.unsubscribe()
+  }
+
+  public importStart(): void {
+    this.loading = true
+  }
+
+  public newActionTriggerOpen(selectedTrigger: ActionTriggerWithActionInfo<KeyboardTriggerData> | undefined): void {
```

Comment:
What the method does and what it is called do not align. From my understanding, this method sets the selected action trigger and tells the component that the form should be rendered in edit mode. So my suggestion would be to rename it to `selectActionTriggerForEditing`.